### PR TITLE
Fix rendering prompt after ExecuteHostCommand

### LINKED ISSUE #755: After `ExecuteHostCommand` is handled, new prompt is rendered
**Platform** Windows 11 WSL2 Ubuntu 20.04,
**Terminal software** wsltty 3.5.1

Describe the problem you are observing.

## Steps to reproduce

```nu
❯ def do_nothing [] {}
❯ let x = {name: do_nothing, modifier: control, keycode: char_g, mode: [vi_normal, vi_insert], event: {send: executehostcommand, cmd: 'do_nothing'}}
❯ $env.config.keybindings = ($env.config.keybindings | append $x)
❯ # type C-g here, then new prompt is rendered
❯
```

### PR DESCRIPTION
**Pull request by kit494way**
Fix #755

### CODE REVIEW

**Comment by fdncred** [comment]
Do you think this PR will fix this nushell issue? https://github.com/nushell/nushell/issues/11981

**Comment by kit494way** [comment]
Yes, I think this PR will fix the issue.

File: src/engine.rs [hunk 1]
```diff
@@ -109,6 +109,9 @@ pub struct Reedline {
     history_cursor_on_excluded: bool,
     input_mode: InputMode,
 
+    // ExecuteHostCommand
+    execute_host_command: bool,
+
```
Comment (sholderbach) [inline src/engine.rs:114]: Small nit: This naming and comment goes a bit lost among all the other configuration focussed members of `Reedline`

It describes a temporary state of `Reedline` so maybe `executing_host_command`/`after_host_command` is more explicit. I don't have a strong opinion where it should be placed in the overall struct, but there we could also emphasize its role.

```suggestion
    // Yielded to the host program after a `ReedlineEvent::ExecuteHostCommand`, thus redraw in-place
    executing_host_command: bool,

```

**Review by sholderbach — APPROVED** [review]
Logic looks good.
From a maintainability perspective a small comment but we should test this with Nushell

**Comment by kit494way** [comment]
Thanks, I accepted the suggestion and renamed.

**Comment by fdncred** [comment]
yay! i was hoping we'd get this in before the next release. i'm wondering what impacts it will have on people's keybindings, meaning, if people are relying on a newline, it could be breaking something. let's find out!

**Pull request merged**